src/imgui/ImGuiDebugger.cc

```diff
@@ -628,14 +628,12 @@ void ImGuiDebugger::drawSlots(MSXCPUInterface& cpuInterface, Debugger& debugger)
 						for (auto sub : xrange(4)) {
 							segments[sub] = romBlocks->read(addr + 0x1000 * sub);
 						}
-						if ((segments[0] == segments[1]) && (segments[2] == segments[3])) {
-							if (segments[0] == segments[2]) { // 16kB
-								ImGui::StrCat('R', segments[0]);
-							} else { // 8kB
-								ImGui::StrCat('R', segments[0], '/', segments[2]);
-							}
-						} else { // 4kB
+						if (segments[0] != segments[1]) { // 4kB
 							ImGui::StrCat('R', segments[0], '/', segments[1], '/', segments[2], '/', segments[3]);
+						} else if (segments[0] != segments[2]) { // 8kB
+							ImGui::StrCat('R', segments[0], '/', segments[2]);
+						} else { // 16kB
+							ImGui::StrCat('R', segments[0]);
```

Comment:
Thanks for your PR. But unfortunately it's wrong. Suppose `segments = {0, 0, 1, 2}`, then your code will print it as a 8kB, but it should be 4kB.

The following will work:

    if ((segments[0] != segments[1]) || (segments[2] != segments[3]) {
        ... handle 4kB
    } else if (segments[0] != segments[2]) {
        ... handle 8kB
    } else {
        ... handle 16kB
    }

This is equivalent to the original code (the same tests, just inverted the condition and swapped the then/else branches), so it's equally efficient. But the code looks nicer because it's less deeply indented.

Can you fix your PR? Or else I can make this tweak this evening.

Reply:
Hmm... I missed your example above.

Reply:
I was trying to write a different solution at first, allowing the slots widget to remember the block size so even if it changes to {0, 0, 0, 0} temporarily, it still knows that the block size is 4k (because it was {0, 1, 2, 3} before).

Reply:
When you say "remember" I don't think I'll agree with the implementation. Because then there should also be a part to "forget" (e.g. we switch to a different slot, or the user inserts a different mapper in the same slot (while the MSX is running)). And such "cache-invalidation-code" is often a source of bugs.

The ImGui-idiom is that the GUI should be stateless (as much as possible). And "remembering" (=keeping state) is thus the wrong approach.

If you insist on always displaying according to the bank-size, then you should query what the bank-size is. I believe there is a way to do this. But I'm not 100% sure, and I cannot look it up right now.

Personally I don't mind seeing something like `R2` for a 8kB mapper. It means segment 2 is selected in the whole 16kB page, so twice the same segment in case of a 8kB mapper (or 4 times the same for a 4kB mapper).

When you're debugging your code, you typically already have a very good understanding about which mapper you're using. Therefor I propose to close this PR.

Reply:
But the meaning of R2/2 and R2 are different, since R2/2 mirrors the memory from 0x0000 to 0x2000 and R2 does not. Besides, I can plug two different rom type mappers at the same time, right?

Reply:
> But the meaning of R2/2 and R2 are different ...

I tried (but failed?) to explain that `R2` and `R2/2` have the same meaning, which is: "segment 2 is selected in the whole page (the whole 16kB region)". I do agree that with this notation you cannot tell the bank-size. But IMHO when you're debugging your program you probably already know what the bank-size is. And therefor, personally, I didn't mind that that information isn't also reflected in the notation. (Motivated by the fact that it simplifies the implementation).

> Besides, I can plug two different rom type mappers at the same time, right?
 
Euhm, you can, but not in the same slot. And thus a single page cannot have a mix of different bank-sizes.
I'm not sure I understand what you meant with that remark.

So sure, if you have a mix of mapper sizes (in different slots), you also have to look at the slot number to know the bank-size of the mapper. But this scenario is rare (maybe not worth complicating the code for?).

> I like the idea of quering the bank size.

Here's something to get you started:
* `src/memory/RomInfo.hh:19` `struct RomTypeInfo` this contains the `blockSize`.
* `src/memory/RomInfo.hh:68` `getBlockSize(RomType)` returns the block size for a given `RomType`.
* But I don't yet see how to get from a `MSXDevice` (like you have in `src/imgui/ImGuiDebugger.cc:621` to the correct `RomType`. Maybe that link is missing?

Reply:
I don't know if I'm completely missing the point here, but I think some notation where you can easily and quickly distinguish an 8kB mapper from a 16kB mapper could be useful when you aren't sure about how OpenMSX detected a mapper. Also useful if you're trying to debug a situation where two cartridges with different mappers are interacting in an unexpected way (I know, it's a long shot), since you can just glance at the widget and know what's active.

Reply:
To be clear: I 100% agree with you that the notation `R2/2` is better than `R2`. So if it were _easy_ to implement we would do it for sure. Since this morning I looked at it a bit more, and I believe implementing this will require some extensions.

In my _opinion_, the extra information is often not critical. So therefor I initially choose to not implement it.

The above is of course only _my_ opinion. _You_'re free to have a different opinion. And if you feel strongly about this, and if you're willing to spend time on this, I can support you with extending openMSX so that it will become possible to implement this.

Reply:
I'll try to be clearer. I'm from an UX background, so that colors my thinking. If you have the same notation mean two different things, this adds an unnecessary difficulty for the user to parse the info, which is what's important, right? Since we have 8kB mappers, and we already have Rx/y to mean two different segments selected in the same page (albeit in two different "subpages", if you will), it's weird to have just Rx when for some reason x=y.

To me, it feels like inconsistent behavior. And I'd have to remind myself that, when that happens, and if I'm using an 8kB mapper, the second "subpage" has the same selected segment as the first one. Which would be easily and directly communicated by Rx/y.

I propose all segments in a given page are always explicitly shown when using an 8kB mapper, even if they are both the same. Plus it would look neater. ;)

Reply:
> I tried (but failed?) to explain that `R2` and `R2/2` have the same meaning, which is: "segment 2 is selected in the whole page (the whole 16kB region)".

I understood fine, but they have completely different outcomes when using different bank sizes.

Reply:
So we agree. We only have a different opinion about how sever this "inconsistent" notation is w.r.t. the extra implementation effort required to make it "consistent".